crowbar_framework/app/models/node_object.rb

```diff
@@ -835,6 +840,20 @@ def _remove_elements_from_node(old, new, from_node)
     @attrs_last_saved = @role.default_attributes.deep_dup
 
     Rails.logger.debug("Done saving node: #{@node.name} - #{crowbar_revision}")
+
+    unless @notify_on_save.empty?
+      Rails.logger.debug("Notifying about saved node: #{@node.name}")
+
+      details = {
+        node: @node.name,
+        attributes: @notify_on_save.keys
```

Comment:
I can move to a set, sure. FWIW, in ruby, a set is implemented (behind the scenes) exactly the way I did, so won't be too much different but I guess it's slightly more readable.